nixos/modules/services/hardware/upower.nix

```diff
@@ -54,6 +220,24 @@ in
 
     systemd.packages = [ cfg.package ];
 
+    systemd.services.upower.environment.UPOWER_CONF_FILE_NAME = pkgs.writeTextFile {
+    name = "UPower.conf";
+    text = ''
+      [UPower]
```

Comment:
How about using `lib.generators.toINI`? You can keep the options separately for documentation:

```
text = generators.toINI {} {
  EnableWattsUpPro = cfg.enableWattsUpPro;
  # ...
};
```

`toINI` has sensible defaults, e.g. converting booleans to proper `"true"` and `"false"` strings.

Reply:
Thanks for the review. I didn't know about `toINI`, trying it now.